src/Evolution/Systems/Ccz4/Ricci.cpp

```diff
@@ -26,21 +28,30 @@ void spatial_ricci_tensor(
     const tnsr::ij<DataType, Dim, Frame>& d_field_p) {
   destructive_resize_components(result,
                                 get_size(get<0, 0>(conformal_spatial_metric)));
-  destructive_resize_components(buffer,
+  destructive_resize_components(buffer1,
```

Comment:
The use of the buffers here is sub-optimal. The default return-by-value does 15 memory allocations(!!!). I would handle the buffer in one of two ways:
1. just take a `gsl::span` that you check is the right size and do the non-owning tensor stuff locally
2. have the return-by-value implementation at least allocation a `Variables` (or `TempBuffer`). This still makes it easy for callers of the not_null version to create very suboptimal code, though, so I don't love this approach.

Reply:
Good point about the allocations. What do you mean by "do the non-owning tensor stuff locally?"

Reply:
Is the performance of the return-by-value function important, or needed at all? If no maybe just delete it or add a note to the docs that it's not optimized? Just making sure we're not wasting effort here.

Reply:
I mean you can do
```cpp
for (size_t storage_index; ) {
  tenosr[storage_index].set_data_ref(buffer.get() + storage_index * size, size);
}
```
for the different DataVectors. Would it make more sense to just pass in the needed quantities as `const&` and require they be computed outside? I'm not a huge fan of functions that compute a bunch of side effect things that individually seem like they could be needed elsewhere. Having said all that, I don't know if these quantities are actually used anywhere else in the CCz4 code :)

Reply:
@nilsdeppe 
Making sure I understand this suggestion correctly: you're suggesting to make one big buffer that stores the 3 current temporaries contiguously, then also create 3 Tensors for those temporaries whose components are just pointers into that big buffer at the right spots, all instead of having those three separate buffers? i.e. this would be 1 allocation instead of 15?

I just checked https://github.com/sxs-collaboration/spectre/pull/3759 and it looks like the `Ccz4::TimeDerivative` is the only thing that currently needs this, so I figure I can just go ahead and compute the temporaries there and pass them into an updated Ricci function

Reply:
@nilsvu 
The return-by-value is not needed by CCZ4 currently, but if I understand correctly, it looks to me like my current implementation has the allocations happening in the `void` one, not the return-by-value one. So I think I need to fix the `void` one anyway, and then because the `void` one is only called by `Ccz4::TimeDerivative`, I think I'll just compute the temporaries there and pass them in. Does that sound good?

Reply:
@nilsvu Only the `void` one is used and in only on place currently. I don't mind addressing the buffer allocations done for the return-by-value if that's what is desired.

Reply:
In the new fixup, I went ahead and just required the temporaries be precomputed and passed in